documentation/coding_guidelines.md

```diff
@@ -254,7 +254,28 @@ We currently use [log](https://docs.rs/log/) for logging.
 
 *Unit tests*
 
-Ideally, all code should be unit tested.  Unit test files should be in the same directory as `mod.rs` and their file names should end in `_test.rs`.  A module to be tested should have the test modules annotated with `#[cfg(test)]`.  For example, if in a crate there is a db module, the expected directory structure is as follows:
+Ideally, all code should be unit tested.  Unit tests should ideally be in the same file as the code it is testing though in a distinct module, using the following syntax:
```

Comment:
nit: Maybe change:
.. it is testing though ..
-> 
... it is testing (though in a distinct module), using..